packages/integration-test/src/setup.js

```diff
@@ -0,0 +1,174 @@
+        this.contractsToDeploy = config.contractsToDeploy;
+
+        this.provider = web3.currentProvider;
+        this.accounts = accounts;
+
+        const networks = ['ropsten', 'rinkeby', 'kovan', 'mainnet'];
+        if (networks.includes(this.NETWORK)) {
+            console.log('Configuring integration test for', this.NETWORK, 'network');
+            this.networkId = networkIDs[capitaliseFirstChar(this.NETWORK)];
+            this.getAddresses();
+            this.getContractPromises();
+            this.getTransactionTestingAddresses();
+        } else {
+            throw new AztecError(codes.UNSUPPORTED_NETWORK, {
+                message: 'Network not recognised, please input: `ropsten`, `rinkeby`, `kovan` or `mainnet`',
+                inputNetwork: this.NETWORK,
+            });
+        }
+    }
+
+    /**
+     * @method getAddresses() - get the addresses for the desired contracts to deploy
+     */
+    getAddresses() {
+        try {
+            this.contractAddresses = getContractAddressesForNetwork(this.networkId);
+        } catch (err) {
+            throw new AztecError(codes.NO_CONTRACT_ADDRESSES, {
+                message: 'No contract addresses exist for the given network ID in @aztec/contract-addresses',
+                networkId: this.networkId,
+            });
+        }
+    }
+
+    /**
+     * @method getContractPromises - get the Truffle contracts for desired contracts, with the provider set
+     */
+    getContractPromises() {
+        this.contractPromises = this.contractsToDeploy.map((nameOfContract) => this.getContractPromise(nameOfContract));
+    }
+
+    /**
+     * @method getContractPromise - get an object whose key is the name of a contract, and the value is a function that
+     * returns the Truffle representation of the contract
+     * @param {string} nameOfContract
+     */
+    getContractPromise(nameOfContract) {
+        const contractObject = {};
+        let extractedContractArtifact;
+        let extractedContractAddress;
+
+        try {
+            extractedContractArtifact = contractArtifacts[nameOfContract];
+        } catch (err) {
+            throw new AztecError(codes.NO_CONTRACT_ARTIFACT, {
+                message: 'Requested contract does not exist in @aztec/contract-artifacts',
+                requestedContract: nameOfContract,
+            });
+        }
+
+        try {
+            extractedContractAddress = this.contractAddresses[nameOfContract];
+        } catch (err) {
+            throw new AztecError(codes.NO_CONTRACT_ADDRESS, {
+                message: 'Requested contract does not have a deployed address in @aztec/contract-addresses',
+                requestedContract: nameOfContract,
+            });
+        }
+
+        contractObject[nameOfContract] = async () => {
+            const truffleRepresentation = this.getTruffleContractRepresentation(extractedContractArtifact);
+            return truffleRepresentation.at(extractedContractAddress);
+        };
```

Comment:
I think it might be cleaner to have this function simply return that `async () => {}`, and construct the object you're constructing in caller.

(i.e. in `getContractPromises()`, do something like: 
```
this.contractPromises = this.contractsToDeploy.map((nameOfContract) => ({ [nameOfContract]: this.getContractPromise(nameOfContract)) });
```

)

Reply:
The suggestion will change a bit if you implement my next comment :)